src/librustc_passes/diagnostics.rs

```diff
@@ -310,6 +310,22 @@ loop {
     break;
 }
 ```
+"##,
+
+E0670: r##"
+Rust 2015 does not permit the use of `async fn`.
+
+Example of erroneous code:
+
+```compile_fail,E0670
+async fn foo() {}
+```
+
+You can remove the keyword `async`:
```

Comment:
Perhaps a better suggestion would be to switch to the 2018 edition?

Reply:
I'll include that as well